src/wp-includes/option.php

```diff
@@ -2144,20 +2144,45 @@ function update_network_option( $network_id, $option, $value ) {
 	 */
 	$value = apply_filters( "pre_update_site_option_{$option}", $value, $old_value, $option, $network_id );
 
+	/*
+	 * To get the actual raw old value from the database, any existing pre filters need to be temporarily disabled.
+	 * Immediately after getting the raw value, they are reinstated.
+	 * The raw value is only used to determine whether a value is present in the database. It is not used anywhere
+	 * else, and is not passed to any of the hooks either.
+	 */
+	global $wp_filter;
+
+	$pre_option_hook = is_multisite() ? "pre_site_option_{$option}" : "pre_option_{$option}";
+	if ( has_filter( $pre_option_hook ) ) {
+		$old_filters = $wp_filter[ $pre_option_hook ];
+		unset( $wp_filter[ $pre_option_hook ] );
+		$raw_old_value = is_multisite() ? get_network_option( $network_id, $option ) : get_option( $option );
```

Comment:
```suggestion
		$raw_old_value = get_network_option( $network_id, $option );
```

Would this not fallback to get_option on single site.

Reply:
`get_network_option()` has quite a bit of logic before it calls `get_option()`, though the main issue is that calling `get_network_option()` on single site will mean that we'll have to do the backup/restore for both `pre_site_option_{$option}` *and* `pre_option_{$option}`. Rather than add doing this twice, it seems cleaner to use `is_multisite()` to determine which function to call.